crates/swc_cli/src/commands/compile.rs

```diff
@@ -13,26 +14,75 @@ use swc::{
     config::{Config, Options},
     try_with_handler, Compiler, TransformOutput,
 };
-use swc_common::{sync::Lazy, FilePathMapping, SourceMap};
+use swc_common::{sync::Lazy, FileName, FilePathMapping, SourceMap};
 use walkdir::WalkDir;
 
 /// Configuration option for transform files.
 #[derive(Parser)]
 pub struct CompileOptions {
+    /// Override a config from .swcrc file.
+    #[clap(long)]
+    config: Option<Vec<String>>,
+
     /// Path to a .swcrc file to use
-    #[clap(long = "config")]
+    #[clap(long)]
     config_file: Option<PathBuf>,
 
-    /// Files to compile
-    files: Vec<PathBuf>,
+    /// Filename to use when reading from stdin - this will be used in
+    /// source-maps, errors etc
+    #[clap(long, short = 'f', group = "input")]
+    filename: Option<PathBuf>,
 
-    /// The output directory
+    /// The name of the 'env' to use when loading configs and plugins. Defaults
+    /// to the value of SWC_ENV, or else NODE_ENV, or else development.
+    #[clap(long)]
+    env_name: Option<String>,
+
+    /// List of glob paths to not compile.
+    #[clap(long)]
+    ignore: Option<String>,
+
+    /// Values: true|false|inline|both
+    #[clap(long)]
+    source_maps: Option<String>,
+
+    /// Define the file for the source map.
     #[clap(long)]
+    source_maps_target: Option<String>,
+
+    /// Set sources[0] on returned source map
+    #[clap(long)]
+    source_file_name: Option<String>,
+
+    /// The root from which all sources are relative.
+    #[clap(long)]
+    source_root: Option<String>,
+
+    /// Automatically recompile files on change
+    #[clap(long)]
+    watch: bool,
+
+    /// Compile all input files into a single file.
+    #[clap(long, group = "output")]
+    out_file: Option<PathBuf>,
+
+    /// The output directory
+    #[clap(long, group = "output")]
     out_dir: Option<PathBuf>,
 
     /// Specify specific file extensions to compile.
     #[clap(long)]
     extensions: Option<Vec<String>>,
+
+    /// Files to compile
+    #[clap(group = "input")]
+    files: Vec<PathBuf>,
+    //Flags legacy @swc/cli supports, might need some thoughts if we need support same.
```

Comment:
These are something to revisit gradually. Some may needed, some maybe out-of-scope from cli features.